### PR TITLE
fix(Thema): Use a specific version of Thema as a dependency.

### LINKED ISSUE #367: Thema version updates policy
Stencila Hub [currently uses Thema 1.16](https://github.com/stencila/hub/search?q=thema&unscoped_q=thema).

[eLife developers have been expanding the eLife theme](https://github.com/stencila/thema/pulls?q=is%3Apr+elife) to allow embedding of https://hub.stenci.la/elife/30274/files/preview/article-30274.md (via a custom CNAME) into an iframe.

Thema has reached [2.1.0](https://github.com/stencila/thema/releases/tag/v2.1.0) so it would be good to know how/if/when this can get into Stencila Hub, and further changes can propagate as more of the eLife theme is built.

### PR DESCRIPTION
**Pull request by nokome**
Closes #367. See there for discussionof approach (and alternatives).

Note: In /open the HTML in the iframe is generated by Encoda, including the `<head>`. Therefore the `THEMA_VERSION` is not applied until `setIframeTheme` is called (i.e. when the theme is changed). I propose NOT to address this now, but instead wait for a planned refactoring of /open. Also, Encoda will be updated to Thema@2 very soon anyway.

### CODE REVIEW

File: director/lib/template_context_processors.py [hunk 1]
```diff
@@ -1,26 +1,34 @@
...
+Each function defines variables that will be available to be
+used in templates. Grouped into functions by type of variable.
+Prefer adding a variable to an existing function, rather than
+creating a new function. If you do create a function make sure
+to add it to `TEMPLATES.OPTIONS.context_processors` in `settings.py`.
+"""
 import typing
 
-from django.conf import settings
+from django import conf
```
Comment (beneboy) [inline director/lib/template_context_processors.py:12]: This is fine and it works, but it's just different then everywhere else in the project where we do `from django.conf import settings` as it was before.

Comment (nokome) [reply director/lib/template_context_processors.py:12]: Yep, I know but `settings` clashed with `def settings`. Should I change it to `from django.conf import settings as django_settings` to be a little more consistent while avoiding the clash.

Comment (beneboy) [reply director/lib/template_context_processors.py:12]: Oh true. Nah don't worry about it.

**Comment by giorgiosironi** [comment]
Thanks for addressing this. This is also a major version update of Thema that would release all the eLife update to the theme that happened in the last month or so.

Not sure if the Renovate policies to fully cover #367 are specified in [`package.json`](https://github.com/stencila/hub/blob/2ead51638beadf063fcd3edef383d1a913dfa7e7/package.json#L49-L59) or out of band (e.g. update all minor versions but not major, PRs that have to be approved rather than auto-merged, etc).

**Comment by nokome** [comment]
> Not sure if the Renovate policies to fully cover #367 are specified in package.json or out of band (e.g. update all minor versions but not major, PRs that have to be approved rather than auto-merged, etc).

At present we do not auto-merge any Renovate PRs (although we are considering it for some repos that have high test coverage e.g. https://github.com/stencila/logga, particularly for `devDependencies`). In this repo, we are generally being more careful about dependency updates e.g. checking changelogs and doing local testing. Please feel free to make suggestions if you think any of these policies could be improved or should be documented.

**Pull request merged**